lib/generators/hobo/migration/migrator.rb

```diff
@@ -489,6 +490,15 @@ def format_options(options, type, changing=false)
           end.compact
         end
 
+        def fk_field_options(model, field_name)
+          options = if (foreign_key = model.constraint_specs.find { |fk| field_name == fk.foreign_key.to_s }) && (parent_table = foreign_key.parent_table_name)
+                      parent_columns = connection.columns(parent_table) rescue []
+                      pk_column = parent_columns.find { |column| column.name == "id" } # right now foreign keys assume id is the target
+                      pk_column ? { limit: pk_column.cast_type.limit } : { limit: 8 }
```

Comment:
Do you mean to have `limit: 8` here when you can't find the `pk_column`? I thought you'd go with `limit: 4` when we can't figure it out?
Also, since the ternary is just looking that the assignment from the previous line, I find `if`/`else` works well. And you could DRY up the hash key (and skip the `options` local variable which will save a bunch of indenting :) ) since all that's changing is the value:
```
if (foreign_key = model.constraint_specs.find { |fk| field_name == fk.foreign_key.to_s })
    && (parent_table = foreign_key.parent_table_name)
    && (parent_columns = connection.columns(parent_table) rescue nil)
    && (pk_column = parent_columns.find { |column| column.name == "id" }) # right now foreign keys assume id is the target
      pk_limit = pk_column.cast_type.limit
    end

{ limit: pk_limit || 4 }

Reply:
We want limit 8 if the PK can't be found because it implies that the table does not exist in the DB...so we're going to be creating the table which _will_ default to a bigint

Reply:
Isn't there some sort of FieldSpec that's in memory when Hobo migration runs? I recall this is different from the fields hash that's contained as a model class instance variable. Happy to look in person.

Reply:
From what I could see, it only builds up the field_specs hash on the models (which is a hash of column name => FieldSpec object)

Id be happy to pair on it if you think theres another